### DISCUSSION TITLE
Browser not starting

### DISCUSSION THREAD

**Discussion by Hunv** [AvaloniaUI/Avalonia#15833] [Q&A]
### Describe the bug

Hi,
I have a project with the Browser and Desktop application.
In general I want a Browser application but for easier and faster debugging, I use the desktop application.

I created a view and used the desktop application for debugging. When I was done I wanted to check the browser application but it is not even starting the application in the browser. I see the splashscreen but after it stays black. If I set some breakpoints to the `Main` and `BuildAvaloniaApp` methods in the program.cs of the browser application, it is not even called.
My application itself is not using any exotic things. In general just standard UI controls and EntityFramework for the data. Currently it does not contain much code.

### To Reproduce

I cannot say what steps are required to reproduce. Even if I remove my complete code, it is not starting the browser application.

### Expected behavior

Browser Application starts.

### Avalonia version

11.0.10

### OS

WebAssembly

### Additional context

It doesn't matter which browser is used (tested Chrome, Edge, Vivaldi and Firefox).
When starting the debugging a new browser tab opens but with "about:blank" as page. I saw this is a known issue, so I just browse to the page manually via http://localhost:5000 or https://localhost:5001 (except for Firefox. It directly opens the correct URL).

**Comment by maxkatz6** [comment]
Please start with an empty app from the latest templates.

**Comment by timunie** [comment]
@Hunv 

1. Use Q&A for questions please as this seems to be a question and not an actual bug

2. "and EntityFramework for the data." Afaik this heavily relies on reflection and could be the root cause. Make sure a hello world app runs fine

3. Upload a very minimal sample here so we can even have a chance to help

Thx

**Reply by timunie** [reply]
made it a Q&A for now. If we have a repo, we can still make it a bug report later on if needed.

**Comment by dbriard** [comment]
I also had issues to debug browser app, with the black (or white) screen after the logo.
What helped a lot is to be able to show the exception in the Browser's inspector console.
I had a look at the sample Catalog project here: 
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Browser/Program.cs
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Browser/ControlCatalog.Browser.csproj
and add logging to console

May be the logging should be added by default in the project template? (at least commented)

**Comment by Hunv** [comment]
Thanks for moving this topic. You are right, this is more a Q&A
I tested a bit and tracked it down to the usage of EntityFramework. The browser is not loading the content when I load the database context of my entity framework class.
@timunie : You mentioned "Afaik this heavily relies on reflection and could be the root cause". What do you mean with it? Why it is a problem in the browser but not in the desktop application?

**Reply by maxkatz6** [reply]
> The browser is not loading the content when I load the database context of my entity framework class.

Are you sure that this database is accessible from the browser? Is it a sqlite database? You wouldn't be able to access server side databases from the browser web page.
EntityFramework in general is known to be a pretty terrible with trimmed applications, which is a requirement for the Browser. But it might not be the main problem.

**Reply by Hunv** [reply]
Hi, No it is a MariaDB. But I load settings for the connection from a config file. May that be the problem?

**Reply by maxkatz6** [reply]
You can't use databases like MariaDB from the browser. You have to create a http web server in between your client and database.

**Reply by Hunv** [reply]
Oh thank you! I wasn't aware of that. 
Is there somewhere a document where I can find limitations about the different platforms?

**Reply by maxkatz6** [reply]
It's pretty hard to document, and Microsoft didn't do this either. Or at least I can't find any docs.

In general, it's safe to say that C# Browser apps have the same limitations as any JS Browser app has. Like, no access to file system, no TCP request nor sockets - only HTTP (and websockets), no windowing...

**Comment by monkeyK1n9** [comment]
Hello, if someone is facing this issue, I found a solution.
It is not related to Entity Framework. In fact, I created another project and changed my IDE, ran the build command from the terminal and it worked.

The problem comes from the launch settings, that is my best guess, launching the app with Chrome.

To make it work, I switched to IIS, and or you can remain with the default setting but run on a different browser.

Fun fact, if you leave the "launchBrowser: false" in the launch settings and open the link yourself in your browser, everything is fine.
